### ISSUE TITLE
Can't get cpanm to install Redis-1.991 

### ISSUE THREAD

**Issue by rwillett** [PerlRedis/perl-redis#131]
Hi,

We've been using the cpanm version of Redis-1.991 for some years and it's always installed. However we have been finding issues with it building and installing on VPS systems running Ubuntu 16.04 OpenVZ.

We have an install script that we run to build a new server and it works fine on all are other systems running Ubuntu 16.04 (Esxi, VmWare Fusion, OpenVZ) but the two systems it's failed on are both SSD based VPS systems. 

We can install Redis-1.991 on a Ubuntu 16.04 with a hard disk, but we have never got the package to install on an SSD based system. We have tried with two different VPS providers and we have the same problem. 

We recognise that this sounds bizarre, but we've just tried again and we can install on a hard disk based Ubuntu 16.04 system but it fails on one with an SSD.

We use the same script each time. 

The error we are getting does vary from which machine we use. 

```
cpanm (App::cpanminus) 1.7044 on perl 5.022001 built for x86_64-linux-gnu-thread-multi
Work directory is /root/.cpanm/work/1534524906.22147
You have make /usr/bin/make
You have /usr/bin/wget
You have /bin/tar: tar (GNU tar) 1.28
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching Redis () on cpanmetadb ...
--> Working on Redis
Fetching http://www.cpan.org/authors/id/D/DA/DAMS/Redis-1.991.tar.gz
-> OK
Unpacking Redis-1.991.tar.gz
Entering Redis-1.991
Checking configure dependencies from META.json
Checking if you have Module::Build::Tiny 0.039 ... Yes (0.039)
Checking if you have ExtUtils::MakeMaker 6.63_03 ... Yes (7.04_01)
Configuring Redis-1.991
Running Build.PL
Creating new 'Build' script for 'Redis' version '1.991'
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Test::TCP 1.19 ... Yes (2.19)
Checking if you have IPC::Open3 0 ... Yes (1.18)
Checking if you have IO::String 0 ... Yes (1.08)
Checking if you have Test::Fatal 0 ... Yes (0.014)
Checking if you have Test::Deep 0 ... Yes (1.128)
Checking if you have Test::CPAN::Meta 0 ... Yes (0.25)
Checking if you have Try::Tiny 0 ... Yes (0.30)
Checking if you have Digest::SHA 0 ... Yes (5.95)
Checking if you have Pod::Coverage::TrustPod 0 ... Yes (0.100005)
Checking if you have IO::Socket::Timeout 0.29 ... Yes (0.32)
Checking if you have IPC::Cmd 0 ... Yes (0.92)
Checking if you have File::Spec 0 ... Yes (3.56)
Checking if you have IO::Handle 0 ... Yes (1.35)
Checking if you have Test::More 0.98 ... Yes (1.001014)
Checking if you have Test::SharedFork 0 ... Yes (0.35)
Building and testing Redis-1.991
cp lib/Redis.pm blib/lib/Redis.pm
cp lib/Redis/List.pm blib/lib/Redis/List.pm
cp lib/Redis/Sentinel.pm blib/lib/Redis/Sentinel.pm
cp lib/Redis/Hash.pm blib/lib/Redis/Hash.pm
t/00-compile.t ............ ok
t/01-basic.t .............. ok
t/02-responses.t .......... ok
    # child is ready to test, signal parent to kill our server
    # parent killed pub/sub redis server, signal child to proceed
    # parent waiting for child 22185...
    # now, check wait_for_messages(), should die...
    # CHILD: reconnected (with a 10s timeout)
    # CHILD: is ready to test, signal parent to restart our server
    # PARENT: killed pub/sub redis server, signal child to proceed
    # CHILD: launch wait_for_messages(2), with reconnect...
    # CHILD: reconnected (with a 10s timeout)
    # PARENT: has relaunched the server...
    # CHILD: after 2 sec, nothing yet, retrying
    # CHILD: launch wait_for_messages(2), with reconnect...
    # CHILD: after 2 sec, nothing yet, retrying
    # CHILD: launch wait_for_messages(2), with reconnect...
    # PARENT: waiting for child 22190...
    # CHILD: child received the message
t/03-pubsub.t ............. ok

#   Failed test 'pipeline with embedded error'
#   at t/04-pipeline.t line 42.
# Compared $data->[1][1]
#    got : 'ERR unknown command `OOPS`, with args beginning with: '
# expect : 'ERR unknown command 'OOPS''
# Looks like you failed 1 test of 11.
t/04-pipeline.t ...........
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/11 subtests
t/05-nonblock.t ........... ok
t/06-on-connect.t ......... ok
    # Sleeping 11 seconds, waiting for Redis to timeout...
    # waiting for sleep command
t/07-reconnect.t .......... ok
t/08-unix-socket.t ........ skipped: Define ENV TEST_REDIS_SERVER_SOCK_PATH to test UNIX socket support
t/10-tie-list.t ........... ok
t/11-timeout.t ............ ok
t/20-tie-hash.t ........... ok
t/30-scripts.t ............ ok
t/42-client_cmds.t ........ ok
t/44-no-unicode-bug.t ..... ok
t/50-fork_safe.t .......... ok
t/release-distmeta.t ...... skipped: these tests are for release candidate testing
t/release-pod-coverage.t .. skipped: these tests are for release candidate testing

Test Summary Report
-------------------
t/04-pipeline.t         (Wstat: 256 Tests: 11 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
Files=18, Tests=281, 73 wallclock secs ( 0.19 usr  0.12 sys +  7.86 cusr  2.68 csys = 10.85 CPU)
Result: FAIL
-> FAIL Installing Redis failed. See /root/.cpanm/work/1534524906.22147/build.log for details. Retry with --force to force install it.
```

We are struggling to work out what the problem is, we have also run the tests with  REDIS_DEBUG=1 and we simply end up with too much data. 

The only thing we can see that is different to the 40-50 machines we've built over the years is the use of an SSD which is frankly weird.

Any help welcomed,

Thanks

Rob

**Comment by ppisar** [comment]
This was fixes in https://github.com/PerlRedis/perl-redis/pull/130. Maintainers need to release a new version.

**Comment by dams** [comment]
Released Redis-1.997-TRIAL. Can you double check it fixes this issue, and reopen if needed ?
